File: asion-account/asion-account-server/src/main/java/org/asion/account/service/AccountManagerImpl.java
package org.asion.account.service;

import com.alibaba.dubbo.config.annotation.DubboService;
import org.asion.account.domain.model.Account;
import org.asion.account.domain.application.AccountManager;
import org.asion.account.domain.model.AccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

/**
 * @author Asion.
 * @since 16/8/24.
 */
@Component("accountManager")
@DubboService(interfaceClass = AccountManager.class)
public class AccountManagerImpl implements AccountManager {

    @Autowired
    private AccountRepository accountRepository;

    @Override
    public Account save(Account account) {
        return accountRepository.save(account);
    }

    @Override
    public Account findOne(Long id) {
        return accountRepository.findOne(id);
    }

    @Override
    public void delete(Long id) {
        accountRepository.delete(id);
    }

    @Override
    public Iterable<Account> findAll() {
        return accountRepository.findAll();
    }

    @Override
    public Account login(Account account) {
        return null;
    }
}

File: asion-account/asion-account-server/src/main/java/io/vincent/account/domain/application/AccountManagerImpl.kt
package io.vincent.account.domain.application

import com.alibaba.dubbo.config.annotation.Service
import io.vincent.account.domain.model.Account
import io.vincent.account.domain.model.AccountRepository
import io.vincent.account.domain.service.AccountService
import io.vincent.common.vo.Page
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.*

/**
 * @author Asion.
 * @since 16/8/24.
 */
@Component
//@DubboService(interfaceClass = AccountManager.class)
@Service(version = "1.0.0",
        application = "\${dubbo.application.id}",
        protocol = ["\${dubbo.protocol.id}"],
        registry = ["\${dubbo.registry.id}"]
)
class AccountManagerImpl @Autowired
constructor(private val accountRepository: AccountRepository,
            private val accountService: AccountService) : AccountManager {

    override fun create(account: Account): Account {
        return accountRepository.create(account)
    }

    override fun update(account: Account): Account {
        return accountRepository.update(account)
    }

    override fun findOne(id: Long): Account? {
        return accountRepository.findOne(id)
    }

    override fun findById(id: Long): Optional<Account> {
        return accountRepository.findById(id)
    }

    override fun findByUsename(username: String): Account? {
//        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        return null
    }

    override fun findByMobile(mobile: String): Account? {
//        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        return null
    }

    override fun findByEmail(email: String): Account? {
//        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
        return null
    }

    override fun findPage(pageNumber: Int, pageSize: Int): Page<Account> {
        return accountRepository.findPage(pageNumber, pageSize)
    }

    override fun disable(id: Long): Long? = accountService.disable(id)
}
